guides/cheatsheets/git.md

```diff
@@ -0,0 +1,283 @@
+# Git Cheatsheet
+
+![Git Meme!](assets/images/gitmeme.jpeg)
+
+## **Welcome!**
+
+Version control is the essence of engineering here at the Shed. Git is a free version control system that's responsible for everything GitHub related that happens locally on your computer. This cheatsheet features the most important and commonly used Git commands for easy reference.
+
+## **Cheatsheet includes:**
+    1. Installation & GUI
+    2. Setup
+    3. Setup & Init
+    4. Stage & Snapshot
+    5. Branch & Merge
+    6. Inspect & Compare
+    7. Tracking Path Changes
+    8. Ignoring Patterns
+    9. Share & Update
+    10. Rewrite History
+    11. Temporary Commits
+
+  
+
+## **Installation & GUI**
+
+With platform specific installers for Git, GitHub also provides the ease of staying up-to-date with the latest releases of the command line tool while providing a graphical user interface for day-to-day interaction, review, and repository synchronisation.
+
+You can download Git for Windows and Mac [here](https://desktop.github.com/).
+
+---
+## **SETUP**
+----
+Configuring user information used across all local repositories.
+
+*Using the command below, you can set a name that is identifiable you when review version history*.
+
+```  
+git config --global user.name “[firstname lastname]”
+```
+
+*Set an email address*.
+
+```
+git config --global user.email “[valid-email]”
+```
+
+## **SETUP & INIT**
+
+Below are commands for initializing and cloning repositories.
+
+*Initialise an existing directory as a Git repository*.
+```
+git init
+```
+*Retrieve an entire repository from a hosted location via URL*.
+```
+git clone [url]
+```
+----
+## **STAGE & SNAPSHOT**
+---
+Working with snapshots and the Git staging area.
+
+*Show modified files in working directory, staged for your next commit*.
+```
+git status
+```
+*Add a file as it looks now to your next commit (stage)*.
+```
+git add [file]
+```
+
+*Add everything your next commit*.
+```
+git add *
+```
+*Commit your staged content as a new commit snapshot*.
+```
+git commit -m “[descriptive message]”
+```
+
+*Upload any changes to your local repository to your remote repository*.
+```
+git push
+```
+
+*Upload any changes to your local repository to your remote repository, rewriting the git history (index)*.
+```
+git push -f   !---- USE WITH CAUTION ----!
```

Comment:
Is it worth calling out that this is a force push and that's it's normally discouraged except for certain use cases like Squashing commits?

Further, you can push multiple branches and only force one like so:

`git push origin +issue/1 issue/2`

Where + denotes that the following branch will be force pushed (issue/1).